### DISCUSSION TITLE
What is the difference `rows` and `flatRows`?

### DISCUSSION THREAD

**Discussion by xkomiks** [TanStack/table#4924] [Q&A] [answered]
export interface RowModel<TData extends RowData> {
  rows: Row<TData>[]
  flatRows: Row<TData>[]
  rowsById: Record<string, Row<TData>>
}

**Comment by nktnet1** [comment, marked as answer]
As of v8, see https://tanstack.com/table/v8/docs/guide/row-models 

---

[Row Model Data Structure](https://tanstack.com/table/v8/docs/guide/row-models#row-model-data-structure)

Each row model will provide you the rows in 3 different useful formats:
- `rows` - An array of rows.
- `flatRows` - An array of rows, but all sub-rows are flattened into the top level.
- `rowsById` - An object of rows, where each row is keyed by its id. This is useful for quickly looking up rows by their id with better performance.

```ts
console.log(table.getRowModel().rows) // array of rows
console.log(table.getRowModel().flatRows) // array of rows, but all sub-rows are flattened into the top level
console.log(table.getRowModel().rowsById['row-id']) // object of rows, where each row is keyed by its `id`
```
